Menu/Provider/GroupMenuProvider.php

```diff
@@ -33,13 +34,35 @@ class GroupMenuProvider implements MenuProviderInterface
     private $pool;
 
     /**
-     * @param FactoryInterface $menuFactory
-     * @param Pool             $pool
+     * @var AuthorizationCheckerInterface
      */
-    public function __construct(FactoryInterface $menuFactory, Pool $pool)
-    {
+    private $checker;
+
+    /**
+     * NEXT_MAJOR: Remove default value of AuthorizationCheckerInterface to null.
+     *
+     * @param FactoryInterface              $menuFactory
+     * @param Pool                          $pool
+     * @param AuthorizationCheckerInterface $checker
+     */
+    public function __construct(
+        FactoryInterface $menuFactory,
+        Pool $pool,
+        AuthorizationCheckerInterface $checker = null
+    ) {
         $this->menuFactory = $menuFactory;
         $this->pool = $pool;
+
+        //NEXT_MAJOR: Remove this deprecation block.
+        if (null === $checker) {
+            @trigger_error(
+                'Passing no 3rd argument is deprecated since version 3.x and will be mandatory in 4.0. Pass 
+                Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface as 3rd argument.',
```

Comment:
```diff
-                'Passing no 3rd argument is deprecated since version 3.x and will be mandatory in 4.0. Pass 
-                Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface as 3rd argument.',
+                'Passing no 3rd argument is deprecated since version 3.x and will be mandatory in 4.0. 
+                Pass Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface as 3rd argument.',
```